### PR TITLE
Add local dependencies as source dependency

### PR DESCRIPTION
**Pull request by metalmatze**
This adds local source dependencies as which will create a symlink inside the `vendor` folder to the actual folder on this. These change will make it obsolete to commit changes first, then run `jb update` and then generate new files.

Here is an example from kube-prometheus.

jsonnetfile.json:
```json
{
    "dependencies": [
        {
            "name": "kube-prometheus",
            "source": {
                "local": {
                    "directory": "jsonnet/kube-prometheus"
                }
            },
            "version": ""
        }
    ]
}
``` 
jsonnetfile.lock.json:
```json
{
    "dependencies": [
        {
            "name": "kube-prometheus",
            "source": {
                "local": {
                    "directory": "jsonnet/kube-prometheus"
                }
            },
            "version": ""
        },
        ...
    ]
}
```

/cc @brancz @squat @paulfantom @lilic @s-urbaniak @kakkoyun

### CODE REVIEW

File: cmd/jb/install_test.go [hunk 1]
```diff
@@ -41,26 +40,28 @@ func TestInstallCommand(t *testing.T) {
...
+			Name:                    "OneURL",
+			URLs:                    []string{"github.com/jsonnet-bundler/jsonnet-bundler@v0.1.0"},
 			ExpectedCode:            0,
 			ExpectedJsonnetFile:     []byte(`{"dependencies": [{"name": "jsonnet-bundler", "source": {"git": {"remote": "https://github.com/jsonnet-bundler/jsonnet-bundler", "subdir": ""}}, "version": "v0.1.0"}]}`),
 			ExpectedJsonnetLockFile: []byte(`{"dependencies": [{"name": "jsonnet-bundler", "source": {"git": {"remote": "https://github.com/jsonnet-bundler/jsonnet-bundler", "subdir": ""}}, "version": "080f157c7fb85ad0281ea78f6c641eaa570a582f"}]}`),
+		}, {
+			Name:                    "Relative",
+			URLs:                    []string{"test/jsonnet/foobar"},
+			ExpectedCode:            0,
+			ExpectedJsonnetFile:     []byte(`{"dependencies":null}`),
```
Comment (brancz) [inline cmd/jb/install_test.go]: dependencies shouldn't be null if installing a package

File: cmd/jb/main.go [hunk 2]
```diff
@@ -73,7 +63,7 @@ func Main() int {
 	initCmd := a.Command(initActionName, "Initialize a new empty jsonnetfile")
 
 	installCmd := a.Command(installActionName, "Install all dependencies or install specific ones")
-	installCmdURLs := installCmd.Arg("packages", "URLs to package to install").URLList()
+	installCmdPaths := installCmd.Arg("paths", "paths to packages to install, URLs or file paths").Strings()
```
Comment (brancz) [inline cmd/jb/main.go]: packages was the right terminology here in my opinion, just "URLs" wasn't .. I'd be ok with just a generic "package identifier"

**Comment by metalmatze** [comment]
Tests are fixed now. I didn't pass the working dir to the check if a local dependency's folder exists. Now checking if the full absolute path exists and it works. :ok_hand: 
As a normal user this should have worked before, but the tests are mocking the working dir for a temporary directory.

PTAL

File: cmd/jb/install_test.go [hunk 3]
```diff
@@ -30,7 +29,7 @@ import (
 func TestInstallCommand(t *testing.T) {
 	testcases := []struct {
 		Name                    string
-		URLs                    []*url.URL
+		URLs                    []string
```
Comment (s-urbaniak) [inline cmd/jb/install_test.go]: nit: this changed semantically to `Paths` now, no?

File: cmd/jb/install_test.go [hunk 4]
```diff
@@ -80,6 +81,8 @@ func TestInstallCommand(t *testing.T) {
 }
 
 func jsonnetFileContent(t *testing.T, filename string, content []byte) {
+	t.Helper()
```
Comment (s-urbaniak) [inline cmd/jb/install_test.go]: :+1:

File: cmd/jb/install_test.go (hunk 3 above)
Comment (metalmatze) [reply cmd/jb/install_test.go]: Indeed, this might not yet have been updated in the test :slightly_smiling_face:

File: cmd/jb/main.go (hunk 2 above)
Comment (s-urbaniak) [reply cmd/jb/main.go]: Thought experiment: we could use the [file URI notation](https://en.wikipedia.org/wiki/File_URI_scheme) to differentiate between `http[s]://...` (URL) and paths using `file://...`.

Then we could stick with the `packages` terminology:
```
installCmdURIs := installCmd.Arg("packages", "packages to install using URI notation. Currently, URLs and local file path URIs according to rfc8089 are supported.").Strings()
```

File: README.md [hunk 5]
```diff
@@ -95,7 +95,7 @@ Commands:
   init
     Initialize a new empty jsonnetfile
 
-  install [<packages>...]
+  install [<paths>...]
```
Comment (s-urbaniak) [inline README.md]: see below, if we go the uri way, we could here say `install [<uris>...]`

**Comment by LiliC** [comment]
Tested this and works on OSX as well 🎉

**Comment by brancz** [comment]
lgtm after @s-urbaniak's comments

**Comment by metalmatze** [comment]
One more thing I need to add/fix. If vendor/ with a symlink already exists it would fail. We need to delete a possible folder/symlink before creating the new symlink, so we don't end up with failing conflicts.

**Comment by metalmatze** [comment]
Adresses all comments and fixed the one bug I mentioned. 

PTAL, it's all done from my side and ready to merge.

**Pull request merged**